File: LeetCode/Medium/1481-least-number-of-unique-integers-after-k-removals/1481-least-number-of-unique-integers-after-k-removals.go
func findLeastNumOfUniqueInts(arr []int, k int) int {
    n := len(arr)
    counter := make(map[int]int)
    count := make([]int, n+1)
    ret := 0

    for i := 0; i < n; i++ {
        if _, ok := counter[arr[i]]; ok {
            counter[arr[i]]++
        } else { counter[arr[i]] = 1 }

    }

    for _, v := range counter {
        ret++
        count[v]++
    }

    i := 1

    for k > 0 {
        if count[i] > 0 {
            count[i]--
            ret--
            k -= i
        } else { i++ }
    }

    if k < 0 { return ret + 1 }
    return ret
}

File: LeetCode/Medium/1481-least-number-of-unique-integers-after-k-removals/1481-least-number-of-unique-integers-after-k-removals.py
class Solution:
    def findLeastNumOfUniqueInts(self, arr: List[int], k: int) -> int:
        counter = Counter(arr)
        count = [0] * (len(arr) + 1)
        ret = 0
        
        for v in counter.values():
            count[v] += 1
            ret += 1
        
        i = 1
        
        while k:
            if count[i]:
                count[i] -= 1
                k -= i
                ret -= 1
            else:
                i += 1
            
            if k < 1:
                break
        
        if k < 0:
            return ret + 1
        return ret
            